abi.md

```diff
@@ -86,16 +125,16 @@ module Top(
 Ports of aggregate type shall be scalarized according to the "Aggregate Type
 Lowering" description in the FIRRTL spec.
 
-Ports of ref type shall be lowered to a Verilog macro of the form `` `define
-ref_<circuit name>_<module name>_<portname> <internal path from module>`` in a
-file with name `ref_<circuit name>_<module name>.sv`.  References to aggregates
-will be lowered to a series of references to ground types.  This ABI does not
-specify whether the original aggregate referent is scalarized or not.
+Ports of ref type on public modules shall, for each public module, be lowered to
+a Verilog macro of the form `` `define ref_<module name>_<portname> <internal
```

Comment:
I'm not sure we should drop the circuit name.   More generally, this would require extmodules to be in a separate circuit and we would need to be able to instantiate modules in other circuits.

Reply:
If we have the circuit name -- and no way to declare the circuit extmodules are in -- how does this work? That's why I think dropping circuit name is helpful, but mostly I don't understand what you're saying ("this would require" -- which are you saying requires that--including circuit or not?).

Are you saying you think we should put extmodules in separate circuit (so that we know what circuit-name the ABI will put their macro under? (and compatibility limits use of one compilation unit to "linking" against definitions of that module that use the "expected" circuit name, not any definition?)), or that we don't want to do that?

If so, how does one use a public module (that isn't top) with probes from another FIRRTL circuit?  What does the extmodule look like and how do we find the right ABI bits to use? Thanks!

Reply:
Or maybe an alternate path is to drop the name from the circuit?  Does it really provide any value anymore?

Basically, the question is does the circuit name matter for anything anywhere?  If it does, it seems we should embed it in the ABI so as to avoid name conflicts when one circuit uses names from another.  This obviously doesn't happen now, but it seems like it should.  Multiple public modules provides most of what you need to have libraries in a namespace.    If it doesn't matter, then sure, drop it all over.

The "would require" thing is referring to specifying the circuit extmodules are in, if circuit is an ABI-affecting namespace.

Reply:
Let's drop it. (Do we go a step further and drop circuits entirely? Probably no, but it's an interesting consideration.) If we don't then it's a hint to the compilation on how to prefix or mangle. However, this is weird as it isn't exactly specified in the ABI.

If we instead remove the name then a prefix can be later added as an optional attribute of the circuit and done in a more exact ABI way.

Other questions/design points: can a circuit be nested or should we design with this in mind? This would make circuit more like a namespace and would probably require dropping the top level circuit. This would require circuit names.

Reply:
Circuit names are dropped in [1b7f50b](https://github.com/chipsalliance/firrtl-spec/pull/130/commits/1b7f50badb1376f26543cbb85a0e84132d9c6ca3). This also removes the redundancy between an annotation without a target and an annotation targeting the circuit.